runners/spark/src/main/java/org/apache/beam/runners/spark/translation/BoundedDataset.java

```diff
@@ -98,9 +98,20 @@
   }
 
   @Override
-  public void cache(String storageLevel) {
-    // populate the rdd if needed
-    getRDD().persist(StorageLevel.fromString(storageLevel));
+  @SuppressWarnings("unchecked")
+  public void cache(String storageLevel, Coder<?> coder) {
+    StorageLevel level = StorageLevel.fromString(storageLevel);
+    if (level.equals(StorageLevel.MEMORY_ONLY()) || level.equals(StorageLevel.MEMORY_ONLY_2())) {
```

Comment:
Perhaps the `if` condition could be extracted for readability and indicative why this condition is here to begin with, something like so:

```java
    if (requiresRddSerialization(level)) {
      // if it is memory only reduce the overhead of moving to bytes
      getRDD().persist(level);
    } else {
      ....
    }
```

Reply:
done, changed to avoidRddSerialization